cmd/manager/main.go

```diff
@@ -89,17 +94,24 @@ func podNamespace() string {
 func main() {
 	var (
 		metricsAddr               string
+		certFile                  string
+		keyFile                   string
 		enableLeaderElection      bool
+		secureMetrics             bool
 		probeAddr                 string
 		cachePath                 string
 		operatorControllerVersion bool
 		systemNamespace           string
 		caCertDir                 string
 		globalPullSecret          string
 	)
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8443", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
 	flag.StringVar(&caCertDir, "ca-certs-dir", "", "The directory of TLS certificate to use for verifying HTTPS connections to the Catalogd and docker-registry web servers.")
+	flag.BoolVar(&secureMetrics, "metrics-secure", true,
```

Comment:
If this is a default value to true, would it make more sense to make this something like --disable-metrics-https and make the description state that it's secure by default?

Reply:
If we use `--disable-metrics-https`, then we would need to explicitly set `true` in the code source to say that metrics is secure by default, rather than as the flag's default value. Would `--disable-metrics-https` default to `false`? And if it’s not explicitly disabled, do we assume metrics are secure? 

_(if !siableMetricsHttps then == MetricsSecure .. little confuse right?)_

Here’s how I see it:

- **Is the metrics server secure (`true`|`false`)?**
- **Do we want the metrics server to start secure or not by default?**

I believe we should secure metrics by default. Therefore, the value of the flag should be `true`. This means, if someone wants to start the manager with metrics in an insecure mode, they would need to explicitly set `--metrics-secure=false`.

For this reason, I’d prefer to keep the current flag name as it aligns with the conventions used in tools like Kubebuilder/SDK. While we’ve deviated in some areas (IHMO), staying as close as possible to those conventions may just make the project more intuitive for contributors.

I hope that makes sense. 
But if everybody wants to change the name, it is fine for me too.

Reply:
The issue is that the description of the flag is not accurate (and this applies to kubebuilder, too, which I just looked at per your comment and it uses the same flag and description). It says "if set, then the endpoint is secured" but if it defaults to true, then that's not entirely true--it's also secured when this flag isn't even called. 

As for how my suggestion would work, I was just imagining a boolean field that defaults to false, then if set to true we skip all the https logic. That's typically how boolean flags to disable a default work, i.e. the -k/--insecure flag in curl for disabling https verification. Having a boolean flag that you pass an explicit boolean value to like `--flag=true or --flag=false` is an odd usage pattern that I haven't really seen. You can see we use this pattern in the other 2 flag.BoolVar values in operator-controller, 'leader-elect' and 'version'.

To illustrate what I mean, in the current PR code:
```
# Secure
manager 
manager --metrics-secure

# Intentionally not secure
manager --metrics-secure=false
```
The two secure examples do the same thing, the metrics are secured, so putting the flag in the 2nd case is unnecessary and potentially confusing if a user is expecting the first case to be insecure by default--typically if you have a flag named something like --metrics-secure, the existence of that flag implies you need that flag in order to enable that capability.

Compare that to something like:
```
# Secure by default
manager

# Intentionally not secure
manager --disable-metrics-tls
```

Here the flag name is directly stating what's changing and since it's optional, a user doesn't have to worry about it at all if they want to leave things secure.

Defaulting boolean flags to false is a pretty widespread standard practice because it's better UX. Some more explanation: https://serendipidata.com/posts/naming-guidelines-for-boolean-variables#guideline-2-always-choose-names-that-enable-setting-false-by-default

Reply:
If we follow your suggestion

`manager --disable-metrics-tls`

That means, it MUST be secure metrics be default right

So, that means we have in the code

If --disable-metrics-tls NOT true then ensure metrics secure.

That is confusing.

Reply:
> it's also secured when this flag isn't even called.

Becuase --metrics-secure by default, it is true
We want it secure by default.
If no flag is informed is secure

Reply:
Because `--metrics-secure` defaults to `true`, we ensure security by default.

**If no flag is provided, the service remains secure.**

We do **not** want a situation where someone runs the project and the Metrics Service is **unsecured** simply because they didn't set the flag. By default, it must be protected. Users need to consciously decide and understand the consequences of running it in an unsecure mode.

This is why I disagree with the suggestion to default it to `false`.

Reply:
Hi @tylerslaton 

Thank you for the detailed feedback! I appreciate the clarification and examples you provided. Here's my perspective:

**Flag Description Accuracy**
You’re absolutely right that the previous flag description (“if set, then the endpoint is secured”) was misleading, as the service is secure by default if the flag isn’t explicitly set. To address this, I’ve updated the description to explicitly reflect the default behavior, reducing potential confusion:

```go
flag.BoolVar(&secureMetrics, "metrics-secure", true, "Controls whether the metrics endpoint is served securely via HTTPS. Defaults to true (secure). Use --metrics-secure=false to serve metrics over HTTP (insecure).") 
```

**Defaulting to Secure by Default**
**The default of true prioritizes security without requiring explicit configuration.** This aligns with best practices for secure-by-default design and minimizes the risk of accidental misconfiguration.

> Security by default "means that the [default](https://itlaw.fandom.com/wiki/Default) [configuration settings](https://itlaw.fandom.com/wiki/Configuration_setting) are the most [secure](https://itlaw.fandom.com/wiki/Secure) settings possible, which are not necessarily the most [user friendly](https://itlaw.fandom.com/wiki/User_friendly) settings."
> https://itlaw.fandom.com/wiki/Security_by_default

REF: https://www.ncsc.gov.uk/information/secure-default

WDYT? Could we agree with the approach?
Can we mark it as solved?

Reply:
Thank you for the suggestion! I agree that using --disable-metrics-tls is a clearer and more intuitive approach. It aligns well with secure-by-default principles while making it explicit when someone opts for an insecure configuration

Done so I am closing this one :-)